dbms/src/Functions/FunctionsHashing.h

```diff
@@ -842,6 +890,10 @@ class FunctionAnyHash : public IFunction
         else if (which.isString()) executeString<first>(icolumn, vec_to);
         else if (which.isFixedString()) executeString<first>(icolumn, vec_to);
         else if (which.isArray()) executeArray<first>(from_type, icolumn, vec_to);
+        else if (which.isDecimal32()) executeDecimal<Decimal32, first>(icolumn, vec_to);
```

Comment:
No need for these branches.

Reply:
Only for Decimal128 and UUID?

Reply:
Yes, because for other types we have specialized versions without virtual calls.

But Decimal, UUID and other are more rare cases - and generic implementation is Ok.